src/gt4py/next/iterator/ir_utils/ir_makers.py

```diff
@@ -401,3 +403,64 @@ def _impl(*its: itir.Expr) -> itir.FunCall:
 def map_(op):
     """Create a `map_` call."""
     return call(call("map_")(op))
+
+
+def cartesian_domain(ranges: Dict[Union[common.Dimension, str], Tuple[Any, Any]]) -> itir.FunCall:
+    """
+    >>> str(
+    ...     cartesian_domain(
+    ...         {
+    ...             common.Dimension(value="IDim", kind=common.DimensionKind.HORIZONTAL): (0, 10),
+    ...             common.Dimension(value="JDim", kind=common.DimensionKind.HORIZONTAL): (0, 20),
+    ...         }
+    ...     )
+    ... )
+    'c⟨ IDimₕ: [0, 10), JDimₕ: [0, 20) ⟩'
+    >>> str(cartesian_domain({"IDim": (0, 10), "JDim": (0, 20)}))
+    'c⟨ IDimₕ: [0, 10), JDimₕ: [0, 20) ⟩'
+
+    """
+
+    return call("cartesian_domain")(
+        *[
+            call("named_range")(
+                itir.AxisLiteral(value=d.value, kind=d.kind)
+                if isinstance(d, common.Dimension)
+                else itir.AxisLiteral(value=d),
+                r[0],
+                r[1],
+            )
+            for d, r in ranges.items()
+        ]
+    )
+
+
+# TODO: merge with cartesian_domain using GridType
```

Comment:
```suggestion
def domain(grid_type: common.GridType, ranges: Dict[Union[common.Dimension, str], Tuple[Any, Any]]) -> itir.FunCall:
```
I would propose this signature.